libbpf-tools/syscount.c

```diff
@@ -0,0 +1,1167 @@
+	[427] = "io_uring_register",
+	[428] = "open_tree",
+	[429] = "move_mount",
+	[430] = "fsopen",
+	[431] = "fsconfig",
+	[432] = "fsmount",
+	[433] = "fspick",
+	[434] = "pidfd_open",
+	[435] = "clone3",
+	[437] = "openat2",
+	[438] = "pidfd_getfd",
+};
+size_t syscall_names_x86_64_size = sizeof(syscall_names_x86_64)/sizeof(char*);
+#endif
+
+static void syscall_name(unsigned n, char *buf, size_t size)
+{
+	const char *name = NULL;
+
+	if (n < syscall_names_size)
+		name = syscall_names[n];
+#ifdef __x86_64__
+	else if (n < syscall_names_x86_64_size)
+		name = syscall_names_x86_64[n];
+#endif
+
+	if (name)
+		strncpy(buf, name, size-1);
+	else
+		snprintf(buf, size, "[unknown: %u]", n);
+}
+
+static int list_syscalls(void)
+{
+	const char **list = syscall_names;
+	size_t size = syscall_names_size;
+
+#ifdef __x86_64__
+	if (!size) {
+		size = syscall_names_x86_64_size;
+		list = syscall_names_x86_64;
+	}
+#endif
+
+	for (size_t i = 0; i < size; i++) {
+		if (list[i])
+			printf("%3zd: %s\n", i, list[i]);
+	}
+
+	return (!list || !size);
+}
+
+static int compar_count(const void *dx, const void *dy)
+{
+	__u64 x = ((struct data_t *) dx)->count;
+	__u64 y = ((struct data_t *) dy)->count;
+	return x > y ? -1 : !(x == y);
+}
+
+static int compar_latency(const void *dx, const void *dy)
+{
+	__u64 x = ((struct data_t *) dx)->total_ns;
+	__u64 y = ((struct data_t *) dy)->total_ns;
+	return x > y ? -1 : !(x == y);
+}
+
+static const char *comm(pid_t pid, char *buf, size_t size)
+{
+	char path[32];
+	int fd;
+	int n;
+
+	snprintf(path, sizeof(path), "/proc/%d/comm", pid);
+	fd = open(path, O_RDONLY);
+	if (fd < 0) {
+		if (env.verbose)
+			warn("open: %s: %s", path, strerror(errno));
+		return "[unknown]";
+	}
+
+	n = read(fd, buf, size);
+	if (n < 0) {
+		warn("read: %s\n", strerror(errno));
+		close(fd);
+		return "[unknown]";
+	}
+	close(fd);
+
+	/* proc will terminate name by 0x0A */
+	if (n > 1)
+		buf[n-1] = '\0';
+
+	return buf;
+}
+
+static const char *agg_col(__u32 key, char *buf, size_t size)
+{
+	char comm_buf[16];
+
+	if (env.process) {
+		snprintf(buf, size, "%-6u %-15s",
+				key,
+				comm(key, comm_buf, sizeof(comm_buf)));
+	} else {
+		syscall_name(key, buf, size);
+	}
+	return buf;
+}
+
+static const char *agg_colname(void)
+{
+	return (env.process) ? "PID    COMM" : "SYSCALL";
+}
+
+static const char *time_colname(void)
+{
+	return (env.milliseconds) ? "TIME (ms)" : "TIME (us)";
+}
+
+static void print_latency_header(void)
+{
+	printf("%-22s %8s %16s\n", agg_colname(), "COUNT", time_colname());
+}
+
+static void print_count_header(void)
+{
+	printf("%-22s %8s\n", agg_colname(), "COUNT");
+}
+
+static void print_latency(struct data_t *vals, size_t count)
+{
+	double div = env.milliseconds ? 1000000.0 : 1000.0;
+	char buf[32];
+
+	print_latency_header();
+	for (int i = 0; i < count && i < env.top; i++)
+		printf("%-22s %8llu %16.3lf\n",
+				agg_col(vals[i].key, buf, sizeof(buf)),
+				vals[i].count,
+				vals[i].total_ns / div);
```

Comment:
formatting looks off, same in print_count. Is it just how Github renders it or it's really off?

Reply:
Does this look better?
```
                col = agg_col(vals[i].key, buf, sizeof(buf));
                printf("%-22s %8llu %16.3lf\n", col, vals[i].count,
                                vals[i].total_ns / div);
```
and
```
                col = agg_col(vals[i].key, buf, sizeof(buf));
                printf("%-22s %8llu\n", col, vals[i].count);
```

Reply:
it's not about extracting agg_col call for me, it was about aligning each parameter with start of format string. It might be tabs vs spaces or you are using different indentation for 2nd+ arguments, don't know. But too lazy to pull your PR locally and check for myself :)